File: dsa/java/src/queue/Queue.java
package queue;

import java.util.ArrayList;

public class Queue<E> {
    private ArrayList<E> queue;

    public Queue(){
        this.queue = new ArrayList<E>();
    }

    public void enqueue(E item){
        this.queue.add(item);
    }

    public E dequeue(){
        return this.queue.remove(0);
    }

    public int getSize(){
        return this.queue.size();
    }
}

File: dsa/go/queue/queue.go
package main

import(
    "fmt"
)

type Queue []int

func (q Queue)enqueue(item int)Queue{
    q = append(q, item)
    return q
}

func (q Queue) dequeue() (Queue, int){
    item := q[0]
    q = q[1:]
    return q, item
}

func (q Queue) getLength()int{
	return len(q)
}

func main(){
	var q Queue
	q = q.enqueue(1)
	q = q.enqueue(2)
	q = q.enqueue(3)
	q = q.enqueue(4)
	q = q.enqueue(5)

	fmt.Println(q.getLength())
	q, i := q.dequeue()
	fmt.Println(i)
	q, i = q.dequeue()
	fmt.Println(i)
}
